### ISSUE TITLE
plugin does not work anymore with latest poetry 1.2.0b2

### ISSUE THREAD

**Issue by ThomasChiroux** [tiangolo/poetry-version-plugin#25]
Hi !

I get this error now:

```
AttributeError
  'ProjectPackage' object has no attribute 'set_version'
  at ~/.local/share/pypoetry/venv/lib/python3.10/site-packages/poetry_version_plugin/plugin.py:98 in activate
       94│                 io.write_line(
       95│                     "poetry-version-plugin: Git tag found, setting "
       96│                     f"dynamic version to: {tag}"
       97│                 )
    →  98│                 poetry.package.set_version(tag)
       99│                 return
```

Certainly related to this update in poetry:

https://github.com/python-poetry/poetry/pull/5486

**Comment by StarLightPL** [comment]
confirmed for 12.0b3 and version from `__init__.py`
```poetry build -vvv
Loading configuration file /home/user/.config/pypoetry/config.toml
Loading the poetry-version-plugin plugin
poetry-version-plugin: Using __init__.py file at project/__init__.py for dynamic version
poetry-version-plugin: Setting package dynamic version to __version__ variable from __init__.py: 1.0.78

  Stack trace:

  12  /opt/poetry/venv/lib/python3.7/site-packages/cleo/application.py:329 in run
       327│
       328│             try:
     → 329│                 exit_code = self._run(io)
       330│             except Exception as e:
       331│                 if not self._catch_exceptions:

  11  /opt/poetry/venv/lib/python3.7/site-packages/poetry/console/application.py:188 in _run
       186│         self._load_plugins(io)
       187│
     → 188│         exit_code: int = super()._run(io)
       189│         return exit_code
       190│

  10  /opt/poetry/venv/lib/python3.7/site-packages/cleo/application.py:423 in _run
       421│             io.input.set_stream(stream)
       422│
     → 423│         exit_code = self._run_command(command, io)
       424│         self._running_command = None
       425│

   9  /opt/poetry/venv/lib/python3.7/site-packages/cleo/application.py:465 in _run_command
       463│
       464│         if error is not None:
     → 465│             raise error
       466│
       467│         return event.exit_code

   8  /opt/poetry/venv/lib/python3.7/site-packages/cleo/application.py:446 in _run_command
       444│
       445│         try:
     → 446│             self._event_dispatcher.dispatch(event, COMMAND)
       447│
       448│             if event.command_should_run():

   7  /opt/poetry/venv/lib/python3.7/site-packages/cleo/events/event_dispatcher.py:23 in dispatch
        21│
        22│         if listeners:
     →  23│             self._do_dispatch(listeners, event_name, event)
        24│
        25│         return event

   6  /opt/poetry/venv/lib/python3.7/site-packages/cleo/events/event_dispatcher.py:84 in _do_dispatch
        82│                 break
        83│
     →  84│             listener(event, event_name, self)
        85│
        86│     def _sort_listeners(self, event_name: str) -> None:

   5  /opt/poetry/venv/lib/python3.7/site-packages/poetry/console/application.py:293 in configure_env
       291│
       292│         io = event.io
     → 293│         poetry = command.poetry
       294│
       295│         env_manager = EnvManager(poetry)

   4  /opt/poetry/venv/lib/python3.7/site-packages/poetry/console/commands/command.py:23 in poetry
        21│     def poetry(self) -> Poetry:
        22│         if self._poetry is None:
     →  23│             return self.get_application().poetry
        24│
        25│         return self._poetry

   3  /opt/poetry/venv/lib/python3.7/site-packages/poetry/console/application.py:132 in poetry
       130│             io=self._io,
       131│             disable_plugins=self._disable_plugins,
     → 132│             disable_cache=self._disable_cache,
       133│         )
       134│

   2  /opt/poetry/venv/lib/python3.7/site-packages/poetry/factory.py:108 in create_poetry
       106│         plugin_manager.load_plugins()
       107│         poetry.set_plugin_manager(plugin_manager)
     → 108│         plugin_manager.activate(poetry, io)
       109│
       110│         return poetry

   1  /opt/poetry/venv/lib/python3.7/site-packages/poetry/plugins/plugin_manager.py:71 in activate
        69│     def activate(self, *args: Any, **kwargs: Any) -> None:
        70│         for plugin in self._plugins:
     →  71│             plugin.activate(*args, **kwargs)
        72│
        73│     def _load_plugin_entry_point(self, ep: metadata.EntryPoint) -> None:

  AttributeError

  'ProjectPackage' object has no attribute 'set_version'

  at /opt/poetry/venv/lib/python3.7/site-packages/poetry_version_plugin/plugin.py:77 in activate
       73│                                     "poetry-version-plugin: Setting package "
       74│                                     "dynamic version to __version__ "
       75│                                     f"variable from __init__.py: {version}"
       76│                                 )
    →  77│                                 poetry.package.set_version(version)
       78│                                 return
       79│             message = (
       80│                 "poetry-version-plugin: No valid __version__ variable found "
       81│                 "in __init__.py, cannot extract dynamic version"```

**Comment by tiangolo** [comment]
Thanks! This was solved in https://github.com/tiangolo/poetry-version-plugin/pull/28 available in version 0.2.0, just released 🎉
